remote/gogs/helper.go

```diff
@@ -62,13 +62,28 @@ func buildFromPush(hook *pushHook) *model.Build {
 		hook.Repo.URL,
 		fixMalformedAvatar(hook.Sender.Avatar),
 	)
+
+	var eventType string
+	var message string
+
+	switch {
+	case hook.RefType == "tag":
+		eventType = model.EventTag
+		message = "Tag " + hook.Ref
+	case hook.RefType == "branch":
+		eventType = ""
+	default:
```

Comment:
setting `eventType = ""` seems a bit undesirable because I'm not sure what the side effect will be. So more from a readability / understandability issue I'd like to avoid this.

Instead I'd recommend we do something like this:

```
func (c *client) Hook(r *http.Request) (*model.Repo, *model.Build, error) {
	var (
		err   error
		repo  *model.Repo
		build *model.Build
	)

	switch r.Header.Get("X-Gogs-Event") {
	case "push":
		var push *pushHook
		push, err = parsePush(r.Body)
		if err == nil && push.RefType != "branch" {
			repo = repoFromPush(push)
			build = buildFromPush(push)
		}
	}
	return repo, build, err
}
```